File: frontend/lib/models/department.dart
class Department {
  final int id;
  final String name;
  final int? supervisorId;
  final String? supervisorName;
  final String createdAt;

  Department({
    required this.id,
    required this.name,
    this.supervisorId,
    this.supervisorName,
    required this.createdAt,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'],
      name: json['name'],
      supervisorId: json['supervisor_id'],
      supervisorName: json['supervisor_name'],
      createdAt: json['created_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'supervisor_id': supervisorId,
      'supervisor_name': supervisorName,
      'created_at': createdAt,
    };
  }
}

File: backend/models/department.py
from utils.db import db
from datetime import datetime

class Department(db.Model):
    __tablename__ = 'departments'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False, unique=True)
    supervisor_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    # Relationships
    supervisor = db.relationship('User', foreign_keys=[supervisor_id], backref='supervised_department')
    jobs = db.relationship('Job', backref='department', lazy=True, cascade='all, delete-orphan')
    
    def to_dict(self):
        """Convert department to dictionary"""
        return {
            'id': self.id,
            'name': self.name,
            'supervisor_id': self.supervisor_id,
            'supervisor_name': self.supervisor.full_name if self.supervisor else None,
            'created_at': self.created_at.isoformat()
        }
    
    def __repr__(self):
        return f'<Department {self.name}>'
